pyblazing/pyblazing/apiv2/context.py

```diff
@@ -3236,3 +3214,169 @@ def log(self, query, logs_table_name="bsql_logs"):
             self.logs_initialized = True
 
         return self.sql(query)
+
+    def _get_progress_bar_format(self):
+        pbfmt = "Steps Complete {n_fmt}/{total_fmt}|{bar}|{percentage:3.0f}% ({elapsed} elapsed)"
+        return pbfmt
+
+    def _wait_completed_single_node(self, graph):
+        query_complete = False
+        while not query_complete:
+            query_complete = graph.query_is_complete()
+            sleep(0.005)
+
+    def _wait_completed_distributed(self, ctxToken):
+        query_complete = False
+        while not query_complete:
+            dask_futures = []
+            for node in self.nodes:
+                worker = node["worker"]
+                dask_futures.append(
+                    self.dask_client.submit(
+                        getQueryIsComplete, ctxToken, workers=[worker], pure=False
+                    )
+                )
+            workers_is_complete = self.dask_client.gather(dask_futures)
+            query_complete = all(workers_is_complete)  # all workers returned true
+            sleep(0.005)
+
+    def _run_progress_bar_single_node(self, graph):
+        from tqdm import tqdm
+
+        query_complete = False
+
+        progress = graph.get_progress()
```

Comment:
this is already in the loop, so we dont need it outseide the loop

Reply:
@williamBlazing yes i think we need that in order to get the max value for the progress bar

Reply:
ok, got it